source/frontend/src/features/leases/detail/LeasePages/payment/models.ts

```diff
@@ -208,8 +208,36 @@ export class FormLeasePayment {
   }
 }
 
-export class FormLeasePeriodWithCategory extends FormLeasePeriod {
-  category: ApiGen_CodeTypes_LeasePaymentCategoryTypes;
+export class LeasePeriodByCategoryProjection {
```

Comment:
This class now encapsulates the data required to render the Payments by category when variable